policy/common/actor.go

```diff
@@ -30,19 +30,23 @@ type Actors struct {
 	Teams         []string `yaml:"teams"`
 	Organizations []string `yaml:"organizations"`
 
-	// Github repository specific interpolation options
+	// Deprecated: use Permissions with "admin" or "write"
 	Admins             bool `yaml:"admins"`
 	WriteCollaborators bool `yaml:"write_collaborators"`
-}
 
-const (
-	GithubWritePermission = "write"
-	GithubAdminPermission = "admin"
-)
+	// A list of GitHub collaborator permissions that are allowed. Values may
+	// be any of "admin", "maintain", "write", "triage", and "read".
+	//
+	// Each desired permission must be listed explicity. For example, even
+	// though "admin" is a superset of "write" in GitHub, both "admin" and
+	// "write" must be included in the list to allow users with either role.
+	Permissions []string
```

Comment:
I also considered a single `Permission` field that specifies a minimum permission. I think that's nicer in someways, but we need the list behavior to replace the `Admins` and `WriteCollaborators` fields, which worked with exact permissions rather than minimum permissions.

For approval, I'm not sure why you'd want something other than minimum permissions, but I can see some value in only assigning reviews to maintainers and not admins. Unfortunately, both approval and reviewer assignment use the same configuration.

We could also have two fields to support both behaviors, but that seemed too complicated.

Reply:
I ended up implementing a direct permission check: the user or team must be granted an exact permission specified in the list by the repository. This matches current behavior for `admins` and changes the behavior for `write_collaborators` to be consistent with `admins`.

It also means there's no way to request reviews from organization owners by using permissions, unless they are also part of a team or added directly. Since we didn't support this before, it should be fine to continue not supporting it.